connection.js

```diff
@@ -518,18 +519,23 @@ class Connection {
         return { reject: false };  // Ok !
 
     }
+
     // Check challenge entirely, and return reject status + optional error msg.
     //
     checkChallenge(notification) {
 
```

Comment:
i also refactored this part a bit to be able to provide different parameters for each function independently

i tested this to work on beta successfully too